service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/service/EncounterReportBundleService.java

```diff
@@ -52,12 +57,23 @@ private void addServiceProvider(Bundle bundle, Encounter encounter) {
 
     }
 
-    private void addIndividual(Bundle bundle, Encounter encounter) {
+    private void addParticipants(Bundle bundle, Encounter encounter) {
         List<Encounter.EncounterParticipantComponent> participantComponents = encounter.getParticipant();
         for (Encounter.EncounterParticipantComponent participantComponent : participantComponents) {
             bundle.addEntry()
                 .setFullUrl(participantComponent.getIndividualTarget().getIdElement().getValue())
                 .setResource(participantComponent.getIndividualTarget());
         }
     }
+
+    private void addAppointment(Bundle bundle, Encounter encounter) {
+        Appointment appointment = encounter.getAppointmentTarget();
+        if (appointment != null) {
+            bundle.addEntry()
+                .setFullUrl(appointment.getIdElement().getValue())
+                .setResource(appointment);
+            encounter.setAppointment(new Reference(appointment));
+            encounter.setAppointmentTarget(appointment);
```

Comment:
What is the purpose for setting appointmentTarget?

Reply:
there should not be the setting of Appointment at all - I did not spot it after merging :) The appointment itself is taken from encounter :)